File: miraculous-frontend/miraculous-ui/src/dto/track-upload-dto/track-upload-dto.ts
export class TrackUploadDto {

  title: string;
  artistId: number;
  artistLogin: string;
  description: string;
  dateOfRelease: string;
  isFree: boolean;
  price: number;
  size: number;
  length: number;

  constructor(title?: string, artistId?: number, artistLogin?: string, description?: string, dateOfRelease?: string,
              isFree?: boolean, price?: number, size?: number, duration?: number) {
    this.title = title;
    this.artistId = artistId;
    this.artistLogin = artistLogin;
    this.description = description;
    this.dateOfRelease = dateOfRelease;
    this.isFree = isFree;
    this.price = price;
    this.size = size;
    this.length = duration;
  }
}

File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/TrackUploadDto.java
package com.example.miraculousbackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class TrackUploadDto {

    @NotBlank(message = "You must provide title of the track")
    private String title;
    private Long artistId;
    private String artistLogin;
    private String description;
    @NotBlank(message = "You must provide date of release")
    private String dateOfRelease;
    @NotNull(message = "Track must be either in paid or free access")
    private Boolean isFree;
    @NotNull(message = "You must give a price for your track")
    private Double price;
    private Double size;
    private Double length;
}
